streams/src/test/java/org/apache/kafka/streams/processor/internals/StandbyTaskTest.java

```diff
@@ -213,6 +224,177 @@ public void testUpdate() throws IOException {
         assertEquals(new Long(30L + 1L), offsets.get(partition2));
     }
 
+    @Test
+    public void testUpdateWindowed() throws IOException {
```

Comment:
nit: improve name

This test seems to be unnecessary complicated. Why do we need to commit offsets 0, 11, 21 ? We should commit 31 right way, check that nothing was expired, advance to 41 and check that oldest got expired. What do we gain by all the intermediate checks?

Reply:
Fair enough. I was following the pattern of the other test, but not for any good reason.